controlplane/kubeadm/internal/proxy/dial.go

```diff
@@ -92,7 +96,17 @@ func (d *Dialer) DialContext(_ context.Context, _ string, addr string) (net.Conn
 		Name(addr).
 		SubResource("portforward")
 
-	dialer := spdy.NewDialer(d.upgrader, &http.Client{Transport: d.proxyTransport}, "POST", req.URL())
+	spdyDialer := spdy.NewDialer(d.upgrader, &http.Client{Transport: d.proxyTransport}, "POST", req.URL())
+
+	websocketDialer, err := portforward.NewSPDYOverWebsocketDialer(req.URL(), d.restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// First attempt tunneling (websocket) dialer, then fallback to spdy dialer.
+	dialer := portforward.NewFallbackDialer(websocketDialer, spdyDialer, func(err error) bool {
```

Comment:
I assume we can't tell from green e2e tests that websocketDialer worked, right? (it could be that we are always falling back to the spdyDialer, which I assume would lead to a lot of unnecessary stuff being done)

Would it make sense to check it once with Tilt manually? (if not already done)